imc-dispatcher SIGSEGV in dispatcher.newConfigForInMemoryChannel

spotted a `SIGSEGV` in a restarted imc-dispatcher logs during an eventing and EKB e2e test run

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x18baaf0]goroutine 154 [running]:
knative.dev/eventing/pkg/reconciler/inmemorychannel/dispatcher.newConfigForInMemoryChannel({0x20fb130?, 0x400069f710?}, 0x40006751e0)
        /remote-source/knative-eventing/app/pkg/reconciler/inmemorychannel/dispatcher/inmemorychannel.go:235 +0x360
knative.dev/eventing/pkg/reconciler/inmemorychannel/dispatcher.(*Reconciler).reconcile(0x40001bee00, {0x20fb130, 0x400069f710}, 0x40006751e0)
        /remote-source/knative-eventing/app/pkg/reconciler/inmemorychannel/dispatcher/inmemorychannel.go:95 +0x158
knative.dev/eventing/pkg/reconciler/inmemorychannel/dispatcher.(*Reconciler).ObserveKind(0x1?, {0x20fb130?, 0x400069f710?}, 0x40004e63f0?)
        /remote-source/knative-eventing/app/pkg/reconciler/inmemorychannel/dispatcher/inmemorychannel.go:84 +0x24
knative.dev/eventing/pkg/client/injection/reconciler/messaging/v1/inmemorychannel.(*reconcilerImpl).Reconcile(0x40004c2f00, {0x20fb130, 0x400069f6b0}, {0x400082c3c0, 0x29})
        /remote-source/knative-eventing/app/pkg/client/injection/reconciler/messaging/v1/inmemorychannel/reconciler.go:257 +0x29c
knative.dev/pkg/controller.(*Impl).processNextWorkItem(0x400009b3e0)
        /remote-source/knative-eventing/app/vendor/knative.dev/pkg/controller/controller.go:542 +0x374
knative.dev/pkg/controller.(*Impl).RunContext.func3()
        /remote-source/knative-eventing/app/vendor/knative.dev/pkg/controller/controller.go:491 +0x5c
created by knative.dev/pkg/controller.(*Impl).RunContext in goroutine 151
        /remote-source/knative-eventing/app/vendor/knative.dev/pkg/controller/controller.go:489 +0x2b0
```

Code in question:

```go
		if isOIDCEnabled {
			conf.ServiceAccount = &types.NamespacedName{
				Name:      *sub.Auth.ServiceAccountName,
				Namespace: imc.Namespace,
			}
		}
```


Spoke with @creydr about this, and he confirmed it can be that `Auth.ServiceAccountName` is nil, when the OIDC feature flag is switched on, quickly before and the OIDC SA is not reconciled at that time already

--- a/pkg/reconciler/inmemorychannel/dispatcher/inmemorychannel.go
+++ b/pkg/reconciler/inmemorychannel/dispatcher/inmemorychannel.go
@@ -241,7 +241,7 @@ func newConfigForInMemoryChannel(ctx context.Context, imc *v1.InMemoryChannel) (
 		}
 
 		conf.Namespace = imc.Namespace
-		if isOIDCEnabled {
+		if isOIDCEnabled && sub.Auth.ServiceAccountName != nil {
 			conf.ServiceAccount = &types.NamespacedName{
 				Name:      *sub.Auth.ServiceAccountName,
 				Namespace: imc.Namespace,

--- a/pkg/reconciler/inmemorychannel/dispatcher/inmemorychannel.go
+++ b/pkg/reconciler/inmemorychannel/dispatcher/inmemorychannel.go
@@ -241,7 +241,7 @@ func newConfigForInMemoryChannel(ctx context.Context, imc *v1.InMemoryChannel) (
 		}
 
 		conf.Namespace = imc.Namespace
-		if isOIDCEnabled && sub.Auth.ServiceAccountName != nil {
+		if isOIDCEnabled && sub.Auth != nil && sub.Auth.ServiceAccountName != nil {
 			conf.ServiceAccount = &types.NamespacedName{
 				Name:      *sub.Auth.ServiceAccountName,
 				Namespace: imc.Namespace,

